api/v1beta1/cryostat_types.go

```diff
@@ -520,10 +520,20 @@ type ReportsSecurityOptions struct {
 
 // TargetDiscoveryOptions provides configuration options to the Cryostat application's target discovery mechanisms.
 type TargetDiscoveryOptions struct {
-	// When true, the Cryostat application will disable the built-in discovery mechanisms. Defaults to false
+	// When true, the Cryostat application will disable the built-in discovery mechanisms. Defaults to false.
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Disable Built-in Discovery",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
 	BuiltInDiscoveryDisabled bool `json:"builtInDiscoveryDisabled,omitempty"`
+	// List of port names that the Cryostat application should look for in order to consider a target as JMX connectable.
+	// Defaults to consider ports named jfr-jmx.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	DiscoveryPortNames []string `json:"discoveryPortNames"`
+	// List of port numbers that the Cryostat application should look for in order to consider a target as JMX connectable.
+	// Defaults to consider ports numbered 9091.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	DiscoveryPortNumbers []int32 `json:"discoveryPortNumbers"`
```

Comment:
Do we want `omitempty` on these?

Reply:
Oh i actually treated `nil` (do not set env var) and `[]` (set it to `-`) as different cases here. So, if we put `omitempty` here, when the controller writes status to CR, it will "accidentally" remove these fields if empty.

Reply:
Hmm, I wonder if it would just be cleaner to have a separate boolean switch for both, like `OverridePortNames` and `OverridePortNumbers` or something.

Reply:
hmmm... right it does sound as a better alternative. If i remember correctly, there should also be an OCP xDescriptor to hide a field based on another one values.

Reply:
I will update it over the weekends^^

Reply:
> hmmm... right it does sound as a better alternative. If i remember correctly, there should also be an OCP xDescriptor to hide a field based on another one values.

Good idea! I think it's this one: https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#20-fielddependency

Thanks for the great work!